prod/prod_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package prod
+
+import (
+	"context"
+	"crypto/rand"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/network"
+	"github.com/ethersphere/swarm/network/retrieval"
+	"github.com/ethersphere/swarm/network/timeouts"
+	"github.com/ethersphere/swarm/pss"
+	"github.com/ethersphere/swarm/pss/trojan"
+	"github.com/ethersphere/swarm/storage"
+	"github.com/ethersphere/swarm/storage/localstore"
+)
+
+// TestRecoveryHook tests that a timeout in netstore
+// invokes correctly recovery hook
+func TestRecoveryHook(t *testing.T) {
+	// setup recovery hook
+	// verify that hook is correctly invoked
+	ctx := context.TODO()
+
+	handlerWasCalled := false // test variable to check handler funcs are correctly retrieved
+
+	// register first handler
+	testHandler := func(ctx context.Context, targets [][]byte, topic trojan.Topic, payload []byte) (*pss.Monitor, error) {
+		handlerWasCalled = true
+		return nil, nil
+	}
+
+	recoverFunc := NewRecoveryHook(testHandler)
+	// call recoverFunc
+	recoverFunc(ctx, chunk.ZeroAddr)
+
+	// verify the hook has been called
+	if handlerWasCalled != true {
+		t.Fatalf("unexpected result for prod Recover func, expected test variable to have a value of %v but is %v instead", true, handlerWasCalled)
+	}
+
+}
+
+func newMockLocalStore(t *testing.T, tags *chunk.Tags) *localstore.DB {
+	dir, err := ioutil.TempDir("", "swarm-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	baseKey := make([]byte, 32)
+	if _, err = rand.Read(baseKey); err != nil {
+		t.Fatal(err)
+	}
+
+	localStore, err := localstore.New(dir, baseKey, &localstore.Options{Tags: tags})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return localStore
+}
+
+// TestSenderCall verifies that pss send is being called correctly
```

Comment:
maybe i am missing something here, but i do not see `pss.Send` being used anywhere in this test

Reply:
much better, thanks